### PR TITLE
Features/save load template

### PR DESCRIPTION
**Pull request by baktun14**
- Create template tables
- CRUD template 
- List user templates
- Create template page
- Import SDL
- Styling

### CODE REVIEW

File: api/src/routers/userRouter.ts [hunk 1]
```diff
@@ -71,3 +86,72 @@ userRouter.put("/updateSettings", requiredUserMiddleware, express.json(), async
...
 });
+
+userRouter.get("/template/:id", optionalUserMiddleware, async (req: JWTRequest, res) => {
+  const userId = req.auth?.sub;
+  const templateId = req.params.id;
+
+  const template = await getTemplateById(templateId, userId);
+
+  if (!template) {
+    res.status(400).send("Template not found");
```
Comment (Redm4x) [inline api/src/routers/userRouter.ts:97]:
```suggestion
    res.status(404).send("Template not found");
```

File: api/src/routers/userRouter.ts [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+import { deleteTemplate, getTemplateById, getTemplates, saveTemplate, saveTemplateDesc } from "@src/db/templateProvider";
 
 export const userRouter = express.Router();
 
+userRouter.get("/byUsername/:username", async (req, res) => {
+  const username = req.params.username;
+
+  const user = await getUserByUsername(username);
+
+  console.log(user);
```
Comment (Redm4x) [inline api/src/routers/userRouter.ts:14]:
```suggestion
```

File: web/src/components/sdl/EditDescriptionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        <Controller
+          control={control}
+          name={`description`}
+          render={({ field }) => (
+            <Box sx={{ marginTop: ".5rem" }}>
+              <InputLabel sx={{ marginBottom: ".5rem" }}>Description</InputLabel>
+              <TextareaAutosize
+                aria-label="Description"
+                minRows={10}
+                placeholder="Write you're guide on how to use this template here!"
```
Comment (Redm4x) [inline web/src/components/sdl/EditDescriptionForm.tsx:69]:
```suggestion
                placeholder="Write your guide on how to use this template here!"
```

File: web/src/utils/sdl/data.ts [hunk 4]
```diff
@@ -31,7 +31,7 @@ export const persistentStorageTypes = [
 
 export const protoTypes = [
   { id: 1, name: "http" },
-  { id: 2, name: "https" },
+  // { id: 2, name: "https" },
```
Comment (Redm4x) [inline web/src/utils/sdl/data.ts:34]: No longer needed?

File: web/src/utils/sdl/sdlImport.ts [hunk 5]
```diff
@@ -59,7 +62,7 @@ export const importSimpleSdl = yamlJson => {
           id: nanoid(),
           port: expose.port,
           as: expose.as,
-          proto: expose.proto || "http",
+          proto: expose.proto === "tpc" ? expose.proto : "http",
```
Comment (Redm4x) [inline web/src/utils/sdl/sdlImport.ts:65]:
```suggestion
          proto: expose.proto === "tcp" ? expose.proto : "http",
```

File: web/src/utils/sdl/data.ts (hunk 4 above)
Comment (baktun14) [reply web/src/utils/sdl/data.ts:34]: I don't think it's supported by Akash

**Review by Redm4x — APPROVED** [review]

**Pull request merged**